packages/smooth_app/lib/pages/onboarding/country_selector.dart

```diff
@@ -0,0 +1,155 @@
+                  borderRadius: BorderRadius.circular(VERY_LARGE_SPACE),
+                ),
+                filled: true,
+                fillColor: const Color.fromARGB(255, 235, 235, 235),
+              ),
+              items:
+                  _countryList.map<DropdownMenuItem<String>>((Country country) {
+                return DropdownMenuItem<String>(
+                  value: country.name,
+                  child: Text(country.name),
+                );
+              }).toList(),
+              hint: Text(
+                appLocalizations.country_chooser_label,
+                style: const TextStyle(
+                    color: Colors.black,
+                    fontSize: 16,
+                    fontWeight: FontWeight.w600),
+              ),
+              onChanged: (String? value) {
+                setState(() {
+                  if (value != null) {
+                    // TODO(jasmeet): Store pref in _userPreferences
+                    _chosenValue = value;
+                  }
+                });
+              },
+            ),
+          );
+        });
+  }
+
+  List<Country> sanitizeCountriesList(List<Country> localizedCountries) {
+    final List<Country> finalCountriesList = <Country>[];
+    final Map<String, OpenFoodFactsCountry> oFFIsoCodeToCountry =
+        <String, OpenFoodFactsCountry>{};
+    final Map<String, Country> localizedIsoCodeToCountry = <String, Country>{};
+    for (final OpenFoodFactsCountry c in OpenFoodFactsCountry.values) {
+      oFFIsoCodeToCountry.putIfAbsent(c.iso2Code.toLowerCase(), () => c);
+    }
+    for (final Country c in localizedCountries) {
+      localizedIsoCodeToCountry.putIfAbsent(
+          c.countryCode.toLowerCase(), () => c);
+    }
+    for (final String countryCode in oFFIsoCodeToCountry.keys) {
+      final Country? localizedCountry = localizedIsoCodeToCountry[countryCode];
+      if (localizedCountry == null) {
+        // No localization for the country name was found, use English name as
+        // default.
+        String countryName = oFFIsoCodeToCountry[countryCode]
+            .toString()
+            .replaceAll('OpenFoodFactsCountry.', '')
+            .replaceAll('_', ' ');
+        countryName =
+            '${countryName[0].toUpperCase()}${countryName.substring(1).toLowerCase()}';
+        final Country country =
+            Country(name: countryName, countryCode: countryName);
+        finalCountriesList.add(country);
+        continue;
+      }
+      // 'gb' is handled as 'uk' in the backend.
+      if (countryCode == 'gb') {
+        final Country modifiedCountry =
+            Country(name: localizedCountry.name, countryCode: 'uk');
+        finalCountriesList.add(modifiedCountry);
+        continue;
+      }
+      finalCountriesList.add(localizedCountry);
+    }
+    return reorderCountries(finalCountriesList);
+  }
+
+  List<Country> reorderCountries(List<Country> countries) {
```

Comment:
You should store the country index 0 in the preferences at init time: imagine I'm in France, the first suggested country is France - I don't have to select anything, just select "Next" in the welcome page. Therefore you cannot rely only on a `onTap` callback, you need to manage the init case too.

Reply:
Done